contracts/StablePriceOracle.sol

```diff
@@ -38,35 +65,26 @@ contract StablePriceOracle is Ownable, PriceOracle {
     }
 
     /**
-     * @dev Sets rent prices.
-     * @param _rentPrices The price array. Each element corresponds to a specific
-     *                    name length; names longer than the length of the array
-     *                    default to the price of the last element.
+     * @dev Returns the pricing premium in wei.
      */
-    function setPrices(uint[] memory _rentPrices) public onlyOwner {
-        rentPrices = _rentPrices;
-        emit RentPriceChanged(_rentPrices);
+    function premium(string calldata name, uint expires, uint duration) external view returns(uint) {
+        return attoUSDToWei(_premium(name, expires, duration));
     }
 
     /**
-     * @dev Returns the price to register or renew a name.
-     * @param name The name being registered or renewed.
-     * @param duration How long the name is being registered or extended for, in seconds.
-     * @return The price of this renewal or registration, in wei.
+     * @dev Returns the pricing premium in internal base units.
      */
-    function price(string calldata name, uint /*expires*/, uint duration) view external returns(uint) {
-        uint len = name.strlen();
-        if(len > rentPrices.length) {
-            len = rentPrices.length;
-        }
-        require(len > 0);
-        uint priceUSD = rentPrices[len - 1].mul(duration);
+    function _premium(string memory name, uint expires, uint duration) internal view returns(uint) {
+        return 0;
+    }
 
-        // Price of one ether in attodollars
+    function attoUSDToWei(uint amount) internal view returns(uint) {
```

Comment:
Does this have to be private? If external, both smart contract and ens-app can refer to same pricing which makes test data setup a bit easier.

Reply:
I know Maker, at least, is antsy about anything that exposes oracle data directly to any caller. This is just a convenience function for calling the oracle directly, though, which we can do offchain.

Reply:
Can you ask the new oracle provider if this is allowed? I can work around our test by calling the current price at the time of seeding data but the data will be static while frontend keeps fetching dynamic data so we have to redeploy the seed data while developing if the price feed start diverting a lot. If frontend and seed data refer to the same data point (from DummyOracle), that diversion won't happen.

Reply:
Will do, with the next PR to refactor to use the new oracle.